deepchem/models/keras_model.py

```diff
@@ -432,8 +465,28 @@ def fit_generator(self,
         c(self, current_step)
       if self.tensorboard and should_log:
         self._log_scalar_to_tensorboard('loss', batch_loss, current_step)
-      if self.wandb and should_log:
-        wandb.log({'loss': batch_loss}, step=current_step)
+      if self.wandb_logger is not None:
+
+        #Get epoch number
+        dataset_size = len(self.wandb_logger.datasets[self.wandb_logger.train_dataset_name])
+        batches_per_epoch = math.ceil(dataset_size / self.batch_size)
+        epoch_num = current_step // batches_per_epoch
+
+        #Log loss and metrics at specified interval
+        if should_log:
```

Comment:
I'm kinda thinking we should ignore log_frequency for logging during training and log the basics (i.e. epoch, batch_loss) every batch.

And the for validation logging we can give the option to do validation every epoch or every N steps, similar to HuggingFace. 

They have an "evaluation_strategy" argument that can turn on/off validation:
_ https://huggingface.co/transformers/main_classes/trainer.html#trainingarguments

Reply:
Would it be appropriate to ignore the log_frequency of KerasModel?
For validation logging, we would have to identify the validation dataset, or is every dataset not the train dataset a "validation set" by default?

Reply:
Would it be appropriate to ignore the log_frequency of KerasModel?
For validation logging, we would have to identify the validation dataset, or is every dataset not the train dataset a "validation set" by default?

Reply:
- Log every batch during training
- WandbLogger would have a evaluation_strategy param, if it's "steps", the interval would just be log_frequency.